rct229/ruleset_functions/baseline_systems/is_baseline_system_10.py

```diff
@@ -0,0 +1,137 @@
+    """
+     Get either Sys-10 or Not_Sys_10 string output which indicates whether the HVAC system is ASHRAE 90.1 2019 Appendix G system 10 (Heating and Ventilation with electric heating).
+
+     Parameters
+     ----------
+     rmi_b json
+         To evaluate if the hvac system is modeled as either Sys-10 or Not_Sys_10 in the B_RMI.
+     hvac_b_id list
+         The id of the hvac system to evaluate.
+     terminal_unit_id_list
+         List list of terminal unit IDs associated with the HVAC system to be evaluated. These are sent to this function from the master get_baseline_system_types function.
+     zone_id_list list
+         list of zone IDs associated with the HVAC system to be evaluated. These are sent to this function from the master get_baseline_system_types function.
+
+    Returns
+     -------
+     The function returns either Sys-10 or Not_Sys_10 string output which indicates whether the HVAC system is ASHRAE 90.1 2019 Appendix G system 10 (Heating and Ventilation with electric heating).
+    """
+
+    is_baseline_system_10 = HVAC_SYS.UNMATCHED
+
+    hvac_b = find_exactly_one_hvac_system(rmi_b, hvac_b_id)
+
+    # check if the hvac system has the required sub systems for system type 10
+    preheat_system = hvac_b.get("preheat_system")
+    has_required_preheat_sys = (
+        preheat_system is None
+        or preheat_system.get("heating_system_type") is None
+        or preheat_system["heating_system_type"] == HEATING_SYSTEM.NONE
+    )
+
+    heating_system = hvac_b.get("heating_system")
+    has_required_heating_sys = (
+        heating_system is None
+        or heating_system.get("heating_system_type") is None
+        or heating_system["heating_system_type"] == HEATING_SYSTEM.NONE
+    )
+
+    has_required_fan_sys = hvac_b.get("fan_system") is None
+
+    are_sys_10_data_matched = (
+        has_required_preheat_sys
+        and has_required_heating_sys
+        and has_required_fan_sys
+        # sub functions handles missing required sys, and return False.
+        and is_hvac_sys_cooling_type_none(rmi_b, hvac_b_id)
+        and does_each_zone_have_only_one_terminal(rmi_b, zone_id_list)
+        and are_all_terminal_heat_sources_electric(rmi_b, terminal_unit_id_list)
+        and are_all_terminal_cool_sources_none_or_null(rmi_b, terminal_unit_id_list)
+        and do_all_terminals_have_one_fan(rmi_b, terminal_unit_id_list)
+        and are_all_terminal_types_cav_with_none_equal_to_null(
+            rmi_b, terminal_unit_id_list
+        )
+    )
+
+    if are_sys_10_data_matched:
+        is_baseline_system_10 = HVAC_SYS.SYS_10
+        return is_baseline_system_10
+
+    # When the first logic of are_sys_10_data_matched is false
+    has_required_heating_sys = (
+        hvac_b.get("heating_system") is not None
+        or hvac_b["heating_system"].get("heating_system_type") is not None
+        or hvac_b["heating_system"]["heating_system_type"] != HEATING_SYSTEM.NONE
+    )
+
+    has_required_preheat_sys = (
```

Comment:
This should be not Null

I recommend to just use one parameter:
has_preheat_sys.
True means has preheat system
False means does not have required system.

Reply:
Presumably, you'll use the new utility function you will create for this.

Reply:
Yes this is in #891.